code/datums/components/aura_healing.dm

```diff
@@ -94,19 +96,26 @@
 		COOLDOWN_START(src, last_heal_effect_time, HEAL_EFFECT_COOLDOWN)
 
 	var/list/remove_alerts_from = current_alerts.Copy()
-
+	var/list/to_heal = list()
 	var/alert_category = "aura_healing_[REF(src)]"
 
-	for (var/mob/living/candidate in (requires_visibility ? view(range, parent) : range(range, parent)))
+	if(requires_visibility)
+		for(var/mob/living/candidate in view(range, parent))
+			to_heal += candidate
+	else
+		for(var/mob/living/candidate in range(range, parent))
+			to_heal += candidate
+
+	for (var/mob/living/candidate as anything in to_heal)
 		if (!isnull(limit_to_trait) && !HAS_TRAIT(candidate, limit_to_trait))
 			continue
 
-		remove_alerts_from -= candidate
-
-		if (!(candidate in current_alerts))
+		var/candi_ref = WEAKREF(candidate)
+		remove_alerts_from -= candi_ref
+		if (!current_alerts[candi_ref])
 			var/atom/movable/screen/alert/aura_healing/alert = candidate.throw_alert(alert_category, /atom/movable/screen/alert/aura_healing, new_master = parent)
 			alert.desc = "You are being healed by [parent]."
-			current_alerts += candidate
+			current_alerts[candi_ref] = TRUE
```

Comment:
the assoc nature is a good idea, but I'm not convinced the WEAKREF does much here.
we "check" this list over once every 0.3 seconds. That's less then the grace window we require for timers or sleeps, so I don't think it's an issue for it to be hardrefs

Reply:
now the list not clearing on destroy is somethin else but whatever

Reply:
Valid point I was just being paranoid